test/integration_test/common_test.go

```diff
@@ -1939,15 +1939,14 @@ func TestMain(m *testing.M) {
 	var once sync.Once
 	once.Do(func() {
 		doDashboardSetup()
+		if err := setup(); err != nil {
+			log.Error("Setup failed with error: %v", err)
+			os.Exit(1)
+		}
+		exitCode := m.Run()
```

Comment:
So, is TestMain getting invoked multiple times in the same job? If yes, that seems to be a problem since that would mean the same set of tests are running multiple times?